src/toxcore/dht/dht_node.rs

```diff
@@ -123,6 +123,37 @@ impl DhtNode {
             }
         }
     }
+
+    /// convert Dhtnode to PackedNode object based on is_ipv6_enabled flag
+    pub fn to_packed_node(self, is_ipv6_enabled: bool) -> PackedNode {
+        let saddr = if is_ipv6_enabled {
+            match self.saddr_v6 {
```

Comment:
Can we unify this with `get_socket_addr`? Perhaps make `get_socket_addr` return address without `Option`?

Reply:
> Or make this method to return Option.

sounds better